providers/rke2-windows.go

```diff
@@ -0,0 +1,33 @@
+package providers
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const RKE2_WINDOWS = "rke2.windows"
+
+
+func IsRKE2Windows(ctx context.Context, k8sClient kubernetes.Interface) (bool, error) {
+	// Look for nodes that have a Windows specific label
+	if isRKE2, err := IsRKE2(ctx, k8sClient); err != nil || isRKE2 {
```

Comment:
IsRKE2 is going to be called twice. Can we just check if the k8s version has rke2 and it has windows nodes?

Reply:
I told him to use this since it also has a harvester check in it. (which may or may not be important in the future)

Reply:
It will be important in the future.

Reply:
```suggestion
	if isRKE2, err := IsRKE2(ctx, k8sClient); err != nil || !isRKE2 {
```

Reply:
Based on the comment from Denise we should also add a harvester check in here as well

Reply:
added the harvester check for rke2-windows